os/drivers/meminfo/meminfo_drv.c

```diff
@@ -107,7 +130,10 @@ static int heapinfo_ioctl(FAR struct file *filep, int cmd, unsigned long arg)
 		heapinfo_parse_heap(heap, option->mode, option->pid);
 		ret = OK;
 		break;
+#endif
 	default:
+		mdbg("Not supported\n");
+		ret = ENOSYS;
```

Comment:
In this case, sometimes it cannot be handled with fail. Because in many cases, it checks the ioctl return with ERROR.
So I think, then we'd better return -ERRNO and check with negative when it checks.